src/org/zaproxy/zap/extension/websocket/WebSocketMessageDTO.java

```diff
@@ -194,7 +194,25 @@ public String getReadablePayload() throws InvalidUtf8Exception {
 			return "";
 		}
 	}
-	
+
+	/**
+	 * Returns content of {@link WebSocketMessageDTO#payload} directly if it is
+	 * of type {@link String}. Otherwise it tries to convert it by replacing unmapped
+	 * characters.
+	 * @return readable representation of payload
+	 * @throws InvalidUtf8Exception
+	 */
+	public String getPayloadAsString() throws InvalidUtf8Exception {
+		if (payload instanceof String) {
+			return (String) payload;
+		} else if (payload instanceof byte[]){
+
+			return Utf8Util.encodePayloadToUtf8ReplaceCharacters((byte[]) payload);
```

Comment:
It could use the string constructor instead which already has that behaviour (not much advantage in providing a utility method for that IMO).